core/src/main/scala/org/apache/spark/deploy/rest/SubmitRestProtocolMessage.scala

```diff
@@ -46,9 +46,6 @@ private[rest] abstract class SubmitRestProtocolMessage {
   val action: String = messageType
   var message: String = null
 
-  // For JSON deserialization
-  private def setAction(a: String): Unit = { }
-
```

Comment:
This was added from the [beginning](https://github.com/apache/spark/commit/6ec0cdc14390d4dc45acf31040f21e1efc476fc0#diff-fb39e366f633463136727a6b6d5b832fR52) and the comment seems to mean this is used. Shall we keep the existing one?

Reply:
Agree, this might be used by reflection; I'd keep it unless we're very sure

Reply:
OK. I will revert this change.